packages/react/src/Testimonial/Testimonial.test.tsx

```diff
@@ -0,0 +1,516 @@
+  })
+
+  it('supports functional ref for Testimonial.Logo component', () => {
+    const mockRef = jest.fn()
+    const {getByRole} = render(
+      <Testimonial>
+        <Testimonial.Quote>Quote text</Testimonial.Quote>
+        <Testimonial.Name>Name</Testimonial.Name>
+        <Testimonial.Logo ref={mockRef}>
+          <img src="/logo.png" alt="Logo" />
+        </Testimonial.Logo>
+      </Testimonial>,
+    )
+
+    const img = getByRole('img', {name: 'Logo'})
+    expect(mockRef).toHaveBeenCalledWith(img)
+  })
+
+  it('renders quote before media section', () => {
+    const {getByRole, getByText} = render(<DefaultTestimonial />)
+
+    const figure = getByRole('figure')
+    const quote = getByText(mockQuote)
+    const name = getByText(mockName)
+
+    const figureChildren = Array.from(figure.children)
+    const quoteIndex = figureChildren.findIndex(child => child.contains(quote))
+    const nameIndex = figureChildren.findIndex(child => child.contains(name))
+
+    expect(quoteIndex).toBeLessThan(nameIndex)
+  })
+
+  it('renders avatar, logo, and name in media section', () => {
+    const {container, getByRole, getByText} = render(
+      <Testimonial>
+        <Testimonial.Quote>Quote text</Testimonial.Quote>
+        <Testimonial.Avatar src={mockAvatarSrc} alt={mockAvatarAlt} />
+        <Testimonial.Logo>
+          <img src="/logo.png" alt="Logo" />
+        </Testimonial.Logo>
+        <Testimonial.Name>{mockName}</Testimonial.Name>
+      </Testimonial>,
+    )
+
+    const mediaSection = container.querySelector('.Testimonial__media')
+    expect(mediaSection).toBeInTheDocument()
+
+    const avatar = getByRole('img', {name: mockAvatarAlt})
+    const logo = getByRole('img', {name: 'Logo'})
+    const name = getByText(mockName)
+
+    expect(mediaSection).toContainElement(avatar)
+    expect(mediaSection).toContainElement(logo)
+    expect(mediaSection).toContainElement(name)
+  })
+
+  it('uses semantic HTML structure', () => {
+    const {getByRole, getByText} = render(<DefaultTestimonial />)
+
+    expect(getByRole('figure')).toBeInTheDocument()
+    expect(getByText(mockQuote).closest('blockquote')).toBeInTheDocument()
+    expect(getByText(mockName).closest('figcaption')).toBeInTheDocument()
+  })
+
+  it('hides quote mark from screen readers', () => {
+    const {getByText} = render(<DefaultTestimonial />)
+
+    const quoteMark = getByText(leftDoubleQuote)
+    expect(quoteMark).toHaveAttribute('aria-hidden', 'true')
+  })
+
+  it('renders with only quote and name', () => {
+    const {getByRole, getByText} = render(
+      <Testimonial>
+        <Testimonial.Quote>Quote text</Testimonial.Quote>
+        <Testimonial.Name>Name</Testimonial.Name>
+      </Testimonial>,
+    )
+
+    expect(getByRole('figure')).toBeInTheDocument()
+    expect(getByText('Quote text')).toBeInTheDocument()
+    expect(getByText('Name')).toBeInTheDocument()
+  })
+
+  it('handles mixed children types', () => {
```

Comment:
it('filters out children that aren't part of the Testimonial namespace', () => {